### PR TITLE
retreive history + result set management

### PR DESCRIPTION
**Pull request by slavshik**

### CODE REVIEW

**Comment by akrherz** [comment]
@paazmaya are you willing to review and sign off on these?  Otherwise, I am unsure of who we have that will.

**Comment by paazmaya** [comment]
Yes, @akrherz, I requested this to happen https://github.com/kvint/XIFF/issues/2

File: src/org/igniterealtime/xiff/data/Message.as [hunk 1]
```diff
@@ -25,14 +25,13 @@
...
-	import org.igniterealtime.xiff.namespaces.xiff_internal;
-
-	/**
+import org.igniterealtime.xiff.core.EscapedJID;
+import org.igniterealtime.xiff.data.id.IIDGenerator;
+import org.igniterealtime.xiff.data.id.IncrementalGenerator;
+import org.igniterealtime.xiff.data.xhtml.XHTMLExtension;
+import org.igniterealtime.xiff.namespaces.xiff_internal;
+
+/**
```
Comment (paazmaya) [inline src/org/igniterealtime/xiff/data/Message.as:34]: Unnecessary indentation change

File: src/org/igniterealtime/xiff/data/archive/ChatStanza.as [hunk 2]
```diff
@@ -0,0 +1,78 @@
+/**
+ * Created by kvint on 06.11.14.
+ */
+package org.igniterealtime.xiff.data.archive {
```
Comment (paazmaya) [inline src/org/igniterealtime/xiff/data/archive/ChatStanza.as:4]: Please follow the same code style as the existing files has, braces at the next line

File: src/org/igniterealtime/xiff/data/archive/ChatStanza.as [hunk 3]
```diff
@@ -0,0 +1,78 @@
+/**
+ * Created by kvint on 06.11.14.
```
Comment (paazmaya) [inline src/org/igniterealtime/xiff/data/archive/ChatStanza.as:2]: Creation comments not needed, but perhaps a short description for the class

File: src/org/igniterealtime/xiff/data/archive/IArchveExtention.as [hunk 4]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Created by kvint on 06.11.14.
+ */
+package org.igniterealtime.xiff.data.archive {
+	import org.igniterealtime.xiff.data.IExtension;
+
+	public interface IArchveExtention extends IExtension {
```
Comment (paazmaya) [inline src/org/igniterealtime/xiff/data/archive/IArchveExtention.as:7]: How about renaming this to `IArchiveExtension`?

**Comment by paazmaya** [comment]
Does these extensions have any specifications, for example in http://xmpp.org/xmpp-protocols/xmpp-extensions/ ?

**Comment by paazmaya** [comment]
Seems there was one test file, which is nice, but would you @kvint have time to create some more tests for the added extensions, and helpers?

**Comment by slavshik** [comment]
@paazmaya, thanks for review. Yes, extensions have specifications I'll add links into comments. Also I'll add missing tests soon.

**Comment by paazmaya** [comment]
@kvint any updates on this PR?

**Pull request closed without merging**